src/de/mod.rs

```diff
@@ -181,8 +155,10 @@ where
     Deserialize::deserialize(de)
 }
 
-fn reader_to_vec<R: Read>(mut reader: R) -> Result<Vec<u8>> {
-    let length = read_i32(&mut reader)?;
+pub(crate) fn reader_to_vec<R: Read>(mut reader: R) -> Result<Vec<u8>> {
```

Comment:
I opted to leave this one here because pulling from a reader can generate IO errors and `crate::de::Error` has a case for those where `crate::raw::Error` doesn't.  Ideally IO error, parsing error, and deserialization error would all be distinct cases in a larger hierarchy but that'll have to wait for RUST-1406.